andela/ah-jarvis
# #159054009 Implement email verification

#### What does this PR do?
Implement email verification

#### Description of Task to be completed?
- Send a verification email on a successful registration
- Verify account on clicking the sent verification link

#### How should this be manually tested?
- Clone this branch
- Create an account with sendgrid and acquire an apikey. 
- Add the api key to the .env as follows 
` EMAIL_HOST_PASSWORD=YOUR_SENDGRID_API_KEY `

``` 
cd ah-jarvis

create and activate a virtual env 

pip install -r requirements.txt

python manage.py makemigrations

python manage.py migrate

coverage run manage.py test

python manage.py run server

```

- Open your postman and create an account with your valid email as follows

```
{
  "user": {
	"username":"jake",
	"email":"validemail@example.com",
	"password":"Pass1234"
  }
}
```

- Open your email and find an email from `webmaster@localhost`
- Try to verify your account
- Boom! There you go, your account is now verified

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#159054009](https://www.pivotaltracker.com/story/show/159054009)

#### Screenshots if any

##### Email mockup
<img width="1279" alt="email" src="https://user-images.githubusercontent.com/22371596/43630851-28010ef6-970a-11e8-9714-2a3ef2cd6213.png">
